Marlin/src/module/stepper/trinamic.cpp

```diff
@@ -36,7 +36,7 @@
 #include <SPI.h>
 
 enum StealthIndex : uint8_t { STEALTH_AXIS_XY, STEALTH_AXIS_Z, STEALTH_AXIS_E };
-#define _TMC_INIT(ST, STEALTH_INDEX) tmc_init(stepper##ST, ST##_CURRENT, ST##_MICROSTEPS, ST##_HYBRID_THRESHOLD, stealthchop_by_axis[STEALTH_INDEX])
+#define _TMC_INIT(ST, STEALTH_INDEX) tmc_init(stepper##ST, ST##_CURRENT, ST##_MICROSTEPS, ST##_HYBRID_THRESHOLD, ST##_COOLSTEP_SPEED_THRESHOLD, ST##_COOLSTEP_LOWER_LOAD_THRESHOLD, ST##_COOLSTEP_UPPER_LOAD_THRESHOLD, ST##_COOLSTEP_SEUP, ST##_COOLSTEP_SEDN, ST##_COOLSTEP_SEIMIN, stealthchop_by_axis[STEALTH_INDEX])
```

Comment:
My first reaction to this is that I don't like forcing every TMC tmc_init implementation to expose all these arguments, just so the 2209 can use them. That said, I don't have a good recommendation for how to actually avoid that while still handling all of this at compile time.

Reply:
I think we could just ifdef these params inside the 2209 init, I suspect the params to init are only useful for the configs that can be changed via gcode.

Reply:
It's a lot of args, but it's not the wrong approach. The compiler will cull the unnamed arguments in the template functions and should produce code just as if they weren't there. If we wanted to get pedantic about it, we'd follow the common approach and use `TMC_INIT_*` macros for each axis, but it isn't needed for this.

Reply:
I considered separate macros per axis, but that just becomes even more code duplication and more places to change every time a new stepper is added (which seems to be every week or two lately!)

Reply:
So which way should we go? Method signature or shortcut to defines?
Personally I prefer the more maintainable code with less duplication.

Reply:
And yet this is what I ultimately decided to do, because without the different inits all the coolstep defines have to exist.